driver/src/solution_submission/mod.rs

```diff
@@ -120,48 +128,193 @@ impl<'a> StableXSolutionSubmitting for StableXSolutionSubmitter<'a> {
         solution: Solution,
         claimed_objective_value: U256,
     ) -> Result<(), SolutionSubmissionError> {
-        self.contract
-            .submit_solution(
-                batch_index.clone(),
-                solution.clone(),
-                claimed_objective_value,
+        retry_with_gas_price_increase(
+            self.contract,
+            batch_index,
+            solution.clone(),
+            claimed_objective_value,
+            self.gas_price_estimating,
+            60_000_000_000u64.into(),
+        )
+        .map_err(|err| {
+            extract_transaction_receipt(&err)
+                .and_then(|tx| {
+                    let block_number = tx.block_number?;
+                    match self.contract.get_solution_objective_value(
+                        batch_index,
+                        solution,
+                        Some(block_number.into()),
+                    ) {
+                        Ok(_) => None,
+                        Err(e) => Some(SolutionSubmissionError::from(e)),
+                    }
+                })
+                .unwrap_or_else(|| SolutionSubmissionError::Unexpected(err.into()))
+        })
+        .map(|_| ())
+    }
+}
+
+fn retry_with_gas_price_increase(
+    contract: &dyn StableXContract,
+    batch_index: U256,
+    solution: Solution,
+    claimed_objective_value: U256,
+    gas_price_estimating: &dyn GasPriceEstimating,
+    gas_cap: U256,
+) -> Result<(), MethodError> {
+    const INCREASE_FACTOR: u32 = 2;
+    const BLOCK_TIMEOUT: usize = 1;
```

Comment:
Isn't 1 block timeout a bit too aggressive? I'm worried that our tx might not yet even have broadcasted until the next block is mined. Maybe we can start with 2?

In the future it would also be good if this could be a function of time remaining as with e.g. 3 minutes or so we could event try the normal safe estimate first in oder to save gas.

Reply:
I had talked to @bh2smith and if mainnet transactions get mined pretty far apart. We figured that if the block timeout were 2 we wouldn't really retry more than once.

I can bump this up to 2 and if we realize the transaction isn't really getting re-sent than we can try for a different strategy (like one where timeout is in function of time remaining).